rs/restorer_test.go

```diff
@@ -4,31 +4,69 @@ import (
 	"context"
 	"testing"
 
+	restore "github.com/Wondertan/go-ipfs-restore"
 	format "github.com/ipfs/go-ipld-format"
 	"github.com/ipfs/go-merkledag"
 	dstest "github.com/ipfs/go-merkledag/test"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+func NewRestorer() *restorer {
+	ds := dstest.Mock()
+	r := &restorer{
+		dag: ds,
+	}
+	return r
+}
```

Comment:
Constructor has to go after the actual struct in the non-test file. And receive DAGService as a parameter, that should be passed from the test.